### PR TITLE
Fix toDateTime64() for negative floats

### LINKED ISSUE #41686: toDateTime64 returns different output with Float type and int type arguments.
select toDateTime64(-86400 ,9,'UTC') ==  1969-12-31 00:00:00.000000000

select toDateTime64(-86400.00 ,9,'UTC') ==  1970-01-01 00:00:00.000000000 

However, I expect both outputs to be same. 

Version affected : 22.8

### PR DESCRIPTION
**Pull request by rschu1ze**
Resolves #41686

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
toDateTime64() now returns the same output with negative integer and float arguments

### CODE REVIEW

File: src/Functions/FunctionsConversion.h [hunk 1]
```diff
@@ -634,11 +634,8 @@ struct ToDateTime64TransformFloat
         : scale(scale_)
     {}
 
-    inline NO_SANITIZE_UNDEFINED DateTime64::NativeType execute(FromType from, const DateLUTImpl &) const
+    NO_SANITIZE_UNDEFINED DateTime64::NativeType execute(FromType from, const DateLUTImpl &) const
     {
-        if (from < 0)
```
Comment (zvonand) [inline src/Functions/FunctionsConversion.h:639]: Maybe some check like in line 621 would be great here

Comment (rschu1ze) [reply src/Functions/FunctionsConversion.h:639]: Done, tnx.

**Review by evillique — APPROVED** [review]

**Comment by evillique** [comment]
Though failure in 00725_memory_tracking seems quite strange.

**Comment by rschu1ze** [comment]
Thanks for approving.

00725_memory_tracking was fine locally and, looking at the test, the failure seems unrelated.

**Pull request merged**